redsky/tests/test_streams.py

```diff
@@ -0,0 +1,324 @@
+##############################################################################
+#
+# xpdan            by Billinge Group
+#                   Simon J. L. Billinge sb2896@columbia.edu
+#                   (c) 2017 trustees of Columbia University in the City of
+#                        New York.
+#                   All rights reserved
+#
+# File coded by:    Christopher J. Wright (CJ-Wright)
+#
+# See AUTHORS.txt for a list of people who contributed.
+# See LICENSE.txt for license information.
+#
+##############################################################################
+from numpy.testing import assert_allclose
+from streams.core import Stream
+
+import redsky.event_streams as es
+from ..event_streams import dstar
+
+
+def test_map(exp_db, start_uid1):
+    source = Stream()
+
+    def add5(img):
+        return img + 5
+
+    ii = {'img': 'pe1_image'}
+    oi = [('image', {'dtype': 'array', 'source': 'testing'})]
+    dp = es.map(dstar(add5),
```

Comment:
Yea, the `source.bla` syntax is more compact, but for when you get really complex graphs breaking everything out is much more easy to keep track of.